apps/extension/src/core/managers/config.ts

```diff
@@ -1,98 +1,256 @@
+import { v4 as uuidv4 } from "uuid"
 import { EventType } from "window.ai"
 
 import { Storage } from "@plasmohq/storage"
 
 import { PortName } from "~core/constants"
 import { Extension } from "~core/extension"
-import { modelCallers } from "~core/llm"
-import { ModelID } from "~public-interface"
+import { local, modelAPICallers, openrouter } from "~core/llm"
+import { ModelID, isKnownModel } from "~public-interface"
 
 import { BaseManager } from "./base"
 
-const defaultLabel: { [K in ModelID]: string } = {
+export enum AuthType {
+  // Let another site handle all authentication
+  External = "external",
+  // Use an API key
+  APIKey = "key"
+}
+
+const APIKeyURL: { [K in ModelID]: string } = {
```

Comment:
I think we should use the Record type - it was made to be a shorthand for this: `Record<ModelID, string>`

Reply:
good call